sklearn/tree/_tree.pyx

```diff
@@ -2693,15 +3403,16 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                            (n_node_samples < 2 * min_samples_leaf) or
                            (weighted_n_node_samples < min_weight_leaf))
 
+                if not is_leaf:
+                    splitter.node_split(impurity, &split, &n_constant_features)
+                    is_leaf = is_leaf or (split.pos >= end)
+
                 if first:
                     impurity = splitter.node_impurity()
                     first = 0
 
                 is_leaf = is_leaf or (impurity <= MIN_IMPURITY_SPLIT)
 
-                if not is_leaf:
-                    splitter.node_split(impurity, &split, &n_constant_features)
-                    is_leaf = is_leaf or (split.pos >= end)
```

Comment:
This change is apparently causing the classification trees to be different and therefore causes some test failures.